### DISCUSSION TITLE
Shouldn't mFeatureLevel be set on GL desktop as well ?

### DISCUSSION THREAD

**Discussion by teub** [google/filament#6026] [Q&A]
I'm not sure it matters so I have not opened an issue, but I noticed on PC, mFeatureLevel is set to 1 whereas max_texture_image_units is >= 31 and I wonder if that's an actual issue.

What do you think ?

Also shouldn't there be an option to matc to get rid of the warnings when using more than N samplers ?

```
} else if constexpr (BACKEND_OPENGL_VERSION == BACKEND_OPENGL_VERSION_GL) {
        // OpenGL version
        initExtensionsGL();
        if (state.major == 4) {
            if (state.minor >= 1) {
                mShaderModel = ShaderModel::DESKTOP;
            }
            if (state.minor >= 3) {
                // figure out our feature level
                if (gets.max_texture_image_units >= 31) {
                    // cubemap arrays are available as of OpenGL 4.0
                    mFeatureLevel = FeatureLevel::FEATURE_LEVEL_2;
                }
            }
            features.multisample_texture = true;
        }
```

**Comment by romainguy** [comment]
@pixelflinger Looks like feature level is set to 2 when on OpenGL 4.3+ with >= 31 texture units.

**Reply by teub** [reply]
Oh, yes, state.minor is 1 in my case. So maybe it's as it should be ?
I'm on windows + geforce 1080, I was assuming it would run the latest GL version (a google search says it should be 4.6)
Maybe that's another issue completely !

**Reply by pixelflinger** [reply]
Yes feature level 2 requires compute. But most importantly it's still work in progress and not fully implemented.
